src/compositor/lipstickcompositor.cpp

```diff
@@ -83,21 +95,14 @@ LipstickCompositor::LipstickCompositor()
 
     connect(QGuiApplication::clipboard(), SIGNAL(dataChanged()), SLOT(clipboardDataChanged()));
 
-    new LipstickCompositorAdaptor(this);
-
-    QDBusConnection systemBus = QDBusConnection::systemBus();
-    if (!systemBus.registerService("org.nemomobile.compositor")) {
-        qWarning("Unable to register D-Bus service org.nemomobile.compositor: %s", systemBus.lastError().message().toUtf8().constData());
-    }
-    if (!systemBus.registerObject("/", this)) {
-        qWarning("Unable to register object at path /: %s", systemBus.lastError().message().toUtf8().constData());
-    }
-
     m_recorder = new LipstickRecorderManager;
     addGlobalInterface(m_recorder);
     addGlobalInterface(new AlienManagerGlobal);
 
     HwcRenderStage::initialize(this);
+
+    setUpdatesEnabled(false);
```

Comment:
I think checking `if (QWindow::handle())` at setUpdatesEnabled is needed (before calling to platformintegration), so that we can call `setUpdatesEnabled(..)` before the actual platform window has been created.